### PR TITLE
Assert ZooKeeper is not running on backup

### PR DESCRIPTION
**Pull request by timaa2k**
## High-level description

This PR changes the ZooKeeper backup/restore script so that it checks that the ZooKeeper process is not running. Before it only asserted that `dcos-exhibitor` was not running which we suspect to not coincide with the former 100% of the cases.

The change enables getting a better understanding of the failure mode next time the flaky test in the JIRA fails. This is not a fix but a mere improvement of the debug output.

## Corresponding DC/OS tickets (required)

  - [DCOS-55827](https://jira.mesosphere.com/browse/DCOS-55827) TestZooKeeperBackup.test_transaction_log_backup_and_restore if flaky.

**edit by @adamtheturtle : We now have an issue just for this, https://jira.mesosphere.com/browse/DCOS-56097 **

### CODE REVIEW

**Comment by timaa2k** [comment]
@mesosphere-mergebot changelog-not-required The backup/restore script is new in 1.14. The changes will be backported.

**Comment by timaa2k** [comment]
@mesosphere-mergebot label Ready For Review

File: packages/exhibitor/extra/dcos_zk_backup.py [hunk 1]
```diff
@@ -41,12 +41,15 @@ def run_command(cmd: str, verbose: bool) -> None:
...
     """
-    Exhibitor and therefore ZooKeeper is in active state (running).
+    Returns whether the ZooKeeper process that Exhibitor controls is running.
     """
+    zk_pid_file = Path('/var/lib/dcos/exhibitor/zk.pid')
+    zk_pid = int(zk_pid_file.read_text())
     try:
-        run_command('systemctl is-active --quiet dcos-exhibitor', verbose)
+        # Check whether the ZooKeeper that Exhibitor controls is running.
+        run_command('kill -0 {zk_pid}'.format(zk_pid=zk_pid), verbose)
```
Comment (jgehrcke) [inline packages/exhibitor/extra/dcos_zk_backup.py]: Quote the man page here:
```
       If  signal  is  0, then no actual signal is sent, but error checking is
       still performed.
```

Comment (jgehrcke) [reply packages/exhibitor/extra/dcos_zk_backup.py]: (can do that later, or just in backports, or not at all)

File: packages/exhibitor/extra/dcos_zk_backup.py [hunk 2]
```diff
@@ -41,12 +41,15 @@ def run_command(cmd: str, verbose: bool) -> None:
 
 def _is_zookeeper_running(verbose: bool) -> bool:
     """
-    Exhibitor and therefore ZooKeeper is in active state (running).
+    Returns whether the ZooKeeper process that Exhibitor controls is running.
     """
+    zk_pid_file = Path('/var/lib/dcos/exhibitor/zk.pid')
```
Comment (gpaul) [inline packages/exhibitor/extra/dcos_zk_backup.py]: If ZK is already stopped, does the zk.pid file still exist?

Comment (timaa2k) [reply packages/exhibitor/extra/dcos_zk_backup.py]: yes

**Comment by timaa2k** [comment]
@jongiddy @jr0d another review and I could backport this :)

File: packages/exhibitor/extra/dcos_zk_backup.py (hunk 1 above)
Comment (adamtheturtle) [reply packages/exhibitor/extra/dcos_zk_backup.py]: I added this to a backports.

**Pull request merged**